blackduck/Client.py

```diff
@@ -102,6 +103,94 @@ def __init__(
         self.base_url = base_url
         self.session = session or HubSession(base_url, timeout, retries, verify)
         self.session.auth = (auth or BearerAuth)(session=self.session, token=token, username=username, password=password)
+        self.root_resources_dict = None
+
+    def list_resources(self, parent=None):
+        """List resources that can be fetched
+
+        Args:
+            self:
+            parent (dict/json): resource object from prior get_resource invocations.
+                                Defaults to None (for root /api/ base).
+
+        Returns:
+            dict(str -> str): of public resource names to urls
+        """
+        if parent is not None and not isinstance(parent, dict):
+            raise TypeError("parent parameter must be a dict if not None")
+        if not parent:
+            # the root resources are in a different format (name -> href)
+            # compared to (rel, href) pairs in _meta.links
+            if self.root_resources_dict is None:
+                # cache root resources for efficiency
+                resp_json = self.session.get("/api/").json()
+                del resp_json['_meta']
```

Comment:
This is fine, so long as we also save the resource url i.e. `"href": "https://<url>/api/projects"`.